### PR TITLE
doc/cephadm/services: Add package conflict warning to osd.rst

### PR DESCRIPTION
**Pull request by alvinowyong**
The command `ceph orch daemon add osd *<host>*:*<device-path>*` will fail on target host if ceph-osd package is installed on target host. This commit adds warning for "creating new osds" section under cephadm services.

## Contribution Guidelines
- To sign and title your commits, please refer to [Submitting Patches to Ceph](https://github.com/ceph/ceph/blob/main/SubmittingPatches.rst).

- If you are submitting a fix for a stable branch (e.g. "pacific"), please refer to [Submitting Patches to Ceph - Backports](https://github.com/ceph/ceph/blob/master/SubmittingPatches-backports.rst) for the proper workflow.

## Checklist
- Tracker (select at least one)
  - [ ] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [x] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [x] No impact that needs to be tracked
- Documentation (select at least one)
  - [x] Updates relevant documentation
  - [ ] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [x] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
</details>

### CODE REVIEW

File: doc/cephadm/services/osd.rst [hunk 1]
```diff
@@ -165,6 +165,8 @@ There are a few ways to create new OSDs:
 
     ceph orch apply -i spec.yml
 
+.. warning:: When creating new OSD using cephadm, ensure that ceph-osd is not already installed on the target host. If ceph-osd is already installed on that host independently, conflicts may arise in the management and control of the OSD, potentially leading to errors, misconfigurations, or unexpected behavior.
+
```
Comment (anthonyeleven) [inline doc/cephadm/services/osd.rst:169]: When deploying new OSDs with `cephadm`, ensure that the `ceph-osd` package is not already installed on the target host.  If it is installed, conflicts may arise in the management and control of the OSD that may lead to errors or unexpected behavior.

**Review by anthonyeleven — CHANGES_REQUESTED** [review]

**Comment by anthonyeleven** [comment]
Also `doc/cephadm/services: Add package conflict warning to osd.rst`

**Comment by alvinowyong** [comment]
Thanks for the inputs, changes made accordingly.

**Review by anthonyeleven — APPROVED** [review]

**Comment by anthonyeleven** [comment]
The failed test looks unrelated, so I'm merging.  Thanks for contributing!

**Pull request merged**